libs/blocks/aside/aside.js

```diff
@@ -59,23 +119,30 @@ function decorateLayout(el) {
     const video = media.querySelector('video');
     if (video) applyHoverPlay(video);
   }
-  const picture = text?.querySelector('picture');
+  const picture = text?.querySelector('p picture');
   const iconArea = picture ? (picture.closest('p') || createTag('p', null, picture)) : null;
   iconArea?.classList.add('icon-area');
   const foregroundImage = foreground.querySelector(':scope > div:not(.text) img')?.closest('div');
   const bgImage = el.querySelector(':scope > div:not(.text) img')?.closest('div');
+  const foregroundMedia = foreground.querySelector(':scope > div:not(.text) video')?.closest('div');
+  const bgMedia = el.querySelector(':scope > div:not(.text) video')?.closest('div');
   const image = foregroundImage ?? bgImage;
-  if (image && !image.classList.contains('text')) {
+  const asideMedia = foregroundMedia ?? bgMedia ?? image;
+  if (asideMedia && !asideMedia.classList.contains('text')) {
     const isSplit = el.classList.contains('split');
-    image.classList.add(`${isSplit ? 'split-' : ''}image`);
+    asideMedia.classList.add(`${isSplit ? 'split-' : ''}image`);
     if (isSplit) {
-      const position = Array.from(image.parentNode.children).indexOf(image);
+      const position = Array.from(asideMedia.parentNode.children).indexOf(asideMedia);
```

Comment:
You should be able to use the array spread syntax here too:
```suggestion
      const position = [...asideMedia.parentNode.children].indexOf(asideMedia);
```

Some [quick info related to array spread syntax](https://stackoverflow.com/questions/40548213/array-from-vs-spread-syntax#:~:text=The%20difference%20is%20that%20spread,doing%20the%20array%2Dcreation%20bit.)

Reply:
Incorporated the suggested changes